### ISSUE TITLE
Create a fixed point module for the stdlib.

### ISSUE THREAD

**Issue by nfurfaro** [FuelLabs/sway#1807]
The FuelVM has no notion of decimals or fixed-point numbers.
Certain applications will want to make use of fixed-point math, ie: to implement bonding curve token pricing schemes.
We should provide a library to make fixed-point numbers simple, safe and performant to use.

This stack overflow answer has links and summaries for a number of libraries in Solidity. The only one I've used personally is [ABDKMath64x64](https://github.com/abdk-consulting/abdk-libraries-solidity/blob/master/ABDKMath64x64.md)
>10x Better Fixed Point Math in Solidity
Use this library to spend a tenth of the gas on your fixed point math.

TL;DR; 

>/**
* Smart contract library of mathematical functions operating with
* signed 64.64-bit fixed point numbers. Signed 64.64-bit fixed point
* number is basically a simple fraction whose numerator is signed
* 128-bit integer and denominator is 2**64. As long as denominator 
* is always the same, there is no need to store it, thus in Solidity
* signed 64.64-bit fixed point numbers are represented by int128
* type holding only the numerator.
*/
